src/overlays/actors/ovl_En_Knight/z_en_knight.c

```diff
@@ -205,237 +380,4016 @@ ActorInit En_Knight_InitVars = {
+        this->igosCurtainReaction = Rand_ZeroFloat(1.9999f); // random 0 or 1
+    }
+
+    this->neckRotationStepScale = 3;
+    this->shadowAlphaFactor = 180.0f;
+    sfxIndex = sSpawnIndex & 3;
+    sSpawnIndex++;
+    this->walkSfx = sKnightWalkSfx[sfxIndex];
+    this->pauseSfx = sKnightPauseSfx[sfxIndex];
+    this->laughSfx = sKnightLaughSfx[sfxIndex];
+    this->attackSfx = sKnightAttackSfx[sfxIndex];
+    this->damagedSfx = sKnightDamagedSfx[sfxIndex];
+    this->defeatedSfx = sKnightDefeatedSfx[sfxIndex];
+    this->voiceSfx = sKnightVoiceSfx[sfxIndex];
+
+    if (this->actor.params != EN_KNIGHT_PARAM_OTHERS) {
+        // Igos
+        s32 i;
+
+        play->specialEffects = sEnKnightEffects;
+
+        for (i = 0; i < ARRAY_COUNT(sEnKnightEffects); i++) {
+            sEnKnightEffects[i].active = false;
+        }
+
+        sIgosInstance = this;
+        sTargetKnight = NULL;
+        sIgosHeadInstance = NULL;
+        this->actor.shape.rot.y = -0x4000;
+        sYaseInstance = (EnKnight*)Actor_Spawn(&play->actorCtx, play, ACTOR_EN_KNIGHT, 1398.0f, 66.0f, 2730.0f,
+                                               this->actor.shape.rot.x, this->actor.shape.rot.y,
+                                               this->actor.shape.rot.z, EN_KNIGHT_PARAM_OTHERS);
+        sDebuInstance = (EnKnight*)Actor_Spawn(&play->actorCtx, play, ACTOR_EN_KNIGHT, 1398.0f, 66.0f, 2969.0f,
+                                               this->actor.shape.rot.x, this->actor.shape.rot.y,
+                                               this->actor.shape.rot.z, EN_KNIGHT_PARAM_OTHERS);
+
+        // Spawn curtains
+        Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_BOSS_06, 0.0f, 0.0f, 0.0f, 0, 0, 0, 0);
+        Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_BOSS_06, 0.0f, 0.0f, 0.0f, 0, 0, 0, 1);
+    }
+
+    if (CHECK_EVENTINF(EVENTINF_INTRO_CS_WATCHED_IGOS_DU_IKANA) && !Flags_GetClear(play, play->roomCtx.curRoom.num)) {
+        if (this == sIgosInstance) {
+            this->actor.world.pos.x = BREG(27) + 1376.0f;
+            this->actor.world.pos.y = 45.0f;
+            this->actor.world.pos.z = BREG(28) + 2864.0f;
+            EnKnight_SetupIgosSitting(this, play);
+            this->skelAnime.curFrame = 25.0f;
+            this->actor.flags &= ~ACTOR_FLAG_TARGETABLE;
+            sMinibossStartBgmCountdown = KREG(83) + 40;
+        } else {
+            EnKnight_SetupWait(this, play);
+            this->doBgChecks = true;
+            this->swordScale = this->shieldScale = 1.0f;
+            this->actor.gravity = -1.5f;
+            this->csCamFov = 60.0f;
+        }
+
+        play->envCtx.lightSettingOverride = 7;
+    } else {
+        EnKnight_SetupIntroCutscene(this, play);
+        play->envCtx.lightSettingOverride = 0;
+    }
+}
+
+void EnKnight_Destroy(Actor* thisx, PlayState* play) {
+    EnKnight* this = THIS;
+}
+
+void EnKnight_CheckRetreatFromExplosiveOrLight(EnKnight* this, PlayState* play) {
```

Comment:
```suggestion
void EnKnight_CheckRetreat(EnKnight* this, PlayState* play) {
```
Not sure we really need to be super verbose and specific about this, and can probably just shorten it.